apps/tx-builder/src/components/Builder.tsx

```diff
@@ -263,34 +289,46 @@ export const Builder = ({
 
   // set when inputValue is visible
   useEffect(() => {
-    const isVisible = async () => {
-      if (contract) {
-        const method = getContractMethod();
-        setIsValueInputVisible(method?.payable || false);
-      } else {
-        setIsValueInputVisible(true);
-      }
-    };
-
-    isVisible();
-  }, [getContractMethod, contract, services, toInput]);
+    if (showCustomData || !contract) {
+      setIsValueInputVisible(true);
+    } else if (contract) {
+      const method = getContractMethod();
+      setIsValueInputVisible(method?.payable || false);
+    }
+  }, [getContractMethod, showCustomData, contract, services, toInput]);
 
   useEffect(() => {
     if (transactions.length === 0) {
       setReviewing(false);
     }
   }, [transactions]);
 
+  useEffect(() => {
+    if (showCustomData) {
+      setCustomDataValue('');
+      try {
+        const txData = getTxData();
+
+        if (txData) {
+          if (services.web3 && !services.web3.utils.isHexStrict(txData.data as string)) {
```

Comment:
you could use typescript's optional chaining here (and a few other places). Just a small nitpick.